miden-lib/src/transaction/outputs.rs

```diff
@@ -48,27 +48,42 @@ pub fn parse_final_account_stub(elements: &[Word]) -> Result<AccountStub, Accoun
 // NOTES EXTRACTOR
 // ================================================================================================
 
-pub fn notes_try_from_elements(elements: &[Word]) -> Result<OutputNote, NoteError> {
+/// Parse an [OutputNote] given a slice of [Word]s.
+///
+/// The expected format is:
+///
+///     [
+///         NOTE_ID,
+///         NOTE_METADATA,
+///         RECIPIENT,
+///         ASSET_HASH,
+///         [num_assets, _, _, _],
+///         ASSET*,
+///     ]
+pub fn parse_output_note_from_slice(elements: &[Word]) -> Result<OutputNote, NoteError> {
+    // Check there is enough data in the slice so the indexes below won't panic
     if elements.len() < CREATED_NOTE_CORE_DATA_SIZE {
         return Err(NoteError::InvalidStubDataLen(elements.len()));
     }
 
+    // Extract the data with fixed offsets
     let note_id: NoteId = elements[CREATED_NOTE_ID_OFFSET as usize].into();
     let metadata: NoteMetadata = elements[CREATED_NOTE_METADATA_OFFSET as usize].try_into()?;
     let recipient = elements[CREATED_NOTE_RECIPIENT_OFFSET as usize].into();
-    let num_assets = elements[CREATED_NOTE_NUM_ASSETS_OFFSET as usize][0];
     let asset_hash: Digest = elements[CREATED_NOTE_ASSET_HASH_OFFSET as usize].into();
+    let num_assets = elements[CREATED_NOTE_NUM_ASSETS_OFFSET as usize][0];
+
+    let asset_start = CREATED_NOTE_ASSETS_OFFSET as usize;
+    let asset_end = asset_start + (num_assets.as_int() as usize);
 
-    if elements.len()
-        < (CREATED_NOTE_ASSETS_OFFSET as usize + num_assets.as_int() as usize) * WORD_SIZE
```

Comment:
The `* WORD_SIZE` is not necessary here. Since the slice is a `&[WORD]` each index in the slice is already a work.

Another way of checking this is looking how the data is accessed on line 68, there is no `* WORD_SIZE` there.